### PR TITLE
added http request ext

### PR DESCRIPTION
**Pull request by RedMan13**

### CODE REVIEW

File: package-lock.json
Comment (GarboMuffin) [inline package-lock.json:1]: don't commit changes to this file

File: extensions/godslayerakp/http.js [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+
+    // inject comments bc yes
+    // @ts-ignore
+    const oldConvert = runtime._convertForScratchBlocks.bind(runtime);
+    const createComment = blockInfo => ({
+        info: blockInfo,
+        xml: `<label text="${blockInfo.slice(3)}"></label>`
+    });
+    // @ts-ignore
+    runtime._convertForScratchBlocks = (blockInfo, categoryInfo) => {
```
Comment (GarboMuffin) [inline extensions/godslayerakp/http.js:16]: use

blockType: Scratch.BlockType.LABEL,
text: "text"

instead of that insecure mess

File: development/server.js [hunk 2]
```diff
@@ -65,6 +65,6 @@ app.use((req, res) => {
 
 // The port the server runs on matters. The editor only treats port 8000 as unsandboxed.
 const PORT = 8000;
-app.listen(8000, () => {
+app.listen(PORT, () => {
```
Comment (GarboMuffin) [inline development/server.js:68]: move to a separate pr

File: extensions/godslayerakp/http.js [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+
+        /**
+         * no need to install runtime as it comes with Scratch var
+         */
+        constructor() {
+            this.clearAll();
+        }
+        getInfo() {
+            return {
+                id: 'extawebrequestsHTTP',
```
Comment (GarboMuffin) [inline extensions/godslayerakp/http.js:123]: prefix extension ids with something related to username

File: extensions/godslayerakp/http.js [hunk 4]
```diff
@@ -0,0 +1,445 @@
...
+                    request.events.activate('reqFail');
+                    request.events.activate('reqEnd');
+                });
+        }
+    }
+
+    const instance = new WebRequests();
+    Scratch.extensions.register(instance);
+    // @ts-ignore
+    runtime.ext_http = instance;
```
Comment (GarboMuffin) [inline extensions/godslayerakp/http.js:444]: you shouldn't need to do that

Comment (GarboMuffin) [reply extensions/godslayerakp/http.js:444]: is there a good reason those extra blocks can't just be in the extension? there is no precedent for an extension to require another one to be used first

File: extensions/godslayerakp/http.js [hunk 5]
```diff
@@ -0,0 +1,445 @@
...
+                            "video/mp4",
+                            "video/x-ms-wmv",
+                            "video/x-msvideo",
+                            "video/x-flv",
+                            "video/webm"
+                        ],
+                        acceptReporters: true
+                    },
+                    jsTypes: {
+                        items: [
```
[the remark this answers is not in the document]
Comment (RedMan13) [reply extensions/godslayerakp/http.js:309]: wdym

File: extensions/godslayerakp/http.js (hunk 4 above)
Comment (RedMan13) [reply extensions/godslayerakp/http.js:444]: > is there a good reason those extra blocks can't just be in the extension? there is no precedent for an extension to require another one to be used first

yes and no, i put them separately becuase they look ugly and dont rlly fit well into the main extension imo

File: extensions/godslayerakp/http.js (hunk 3 above)
Comment (RedMan13) [reply extensions/godslayerakp/http.js:123]: oh

File: development/server.js (hunk 2 above)
Comment (RedMan13) [reply development/server.js:68]: alr

File: extensions/godslayerakp/http.js (hunk 1 above)
Comment (RedMan13) [reply extensions/godslayerakp/http.js:16]: its not insecure? and this merge was pre-labels but yes it needs changed

Comment (RedMan13) [reply extensions/godslayerakp/http.js:16]: fixed...

File: development/server.js (hunk 2 above)
Comment (RedMan13) [reply development/server.js]: fixed...

File: extensions/godslayerakp/http.js (hunk 3 above)
Comment (RedMan13) [reply extensions/godslayerakp/http.js]: fixed...

File: package-lock.json
Comment (RedMan13) [reply package-lock.json:1]: idk how to fix

**Comment by RedMan13** [comment]
and now its broken :/

File: package-lock.json
Comment (RedMan13) [reply package-lock.json:1]: nvm figured it out i think

File: extensions/godslayerakp/http.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (RedMan13) [reply extensions/godslayerakp/http.js:310]: ima close this since there isnt any reason for it i can see

File: extensions/godslayerakp/http.js (hunk 4 above)
Comment (RedMan13) [reply extensions/godslayerakp/http.js:457]: reading it again i noticed you specified "required", they are not meant to be required and for the most part are not

Comment (GarboMuffin) [reply extensions/godslayerakp/http.js:459]: it seems to me that every block in httpExtra uses ext_http, so yes it is a required dependency

**Comment by GarboMuffin** [comment]
thanks for your patience

does httpExtra do anything that isn't possible with other JSON extensions here?

if it does, is there a good reason we can't just put them in the main extension? perhaps using a button to hide them by default, but it doesn't look like there's too many there anyways

**Comment by RedMan13** [comment]
> thanks for your patience
> 
> does httpExtra do anything that isn't possible with other JSON extensions here?
> 
> if it does, is there a good reason we can't just put them in the main extension? perhaps using a button to hide them by default, but it doesn't look like there's too many there anyways

yes it does do more then what json can as it manualy edits the whole request options object 
a button to toggle it is actualy a rlly good idea

File: extensions/godslayerakp/http.js (hunk 4 above)
Comment (RedMan13) [reply extensions/godslayerakp/http.js]: > it seems to me that every block in httpExtra uses ext_http, so yes it is a required dependency

oh you meant the other way
i was reading it as http requiring httpExtra

**Comment by GarboMuffin** [comment]
thank you for your patience
